omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/PatientAllergyResource2_0.java

```diff
@@ -0,0 +1,238 @@
+		description.addProperty("comment");
+		description.addProperty("reactions");
+		return description;
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getUpdatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		return getCreatableProperties();
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingSubResource#getParent(java.lang.Object)
+	 */
+	@Override
+	public Patient getParent(Allergy instance) {
+		return instance.getPatient();
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingSubResource#setParent(java.lang.Object,
+	 *      java.lang.Object)
+	 */
+	@Override
+	public void setParent(Allergy instance, Patient patient) {
+		instance.setPatient(patient);
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.SubResource#doGetAll(java.lang.Object,
+	 *      org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	public NeedsPaging<Allergy> doGetAll(Patient parent, RequestContext context) throws ResponseException {
+		List<Allergy> allergiesList = new ArrayList<Allergy>();
+		Allergies allergies = new Allergies();
+		if (parent != null) {
+			allergies = Context.getPatientService().getAllergies(parent);
+			if (allergies.getAllergyStatus().equals(Allergies.UNKNOWN)) {
+				// return 404 status for a patient whoose allergy status is unknown
+				throw new ObjectNotFoundException();
+			} else if (allergies.getAllergyStatus().equals(Allergies.NO_KNOWN_ALLERGIES)) {
+				// empty list
+			} else if (allergies.getAllergyStatus().equals(Allergies.SEE_LIST)) {
+				// return list
+				allergiesList.addAll(allergies);
+			}
+		}
+		return new NeedsPaging<Allergy>(allergiesList, context);
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getByUniqueId(java.lang.String)
+	 */
+	@Override
+	public Allergy getByUniqueId(String uuid) {
+		return Context.getPatientService().getAllergyByUuid(uuid);
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#delete(java.lang.Object,
+	 *      java.lang.String, org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	public void delete(Allergy allergy, String reason, RequestContext context) throws ResponseException {
+		Context.getPatientService().voidAllergy(allergy, reason);
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#delete(java.lang.Object,
+	 *      java.lang.String, java.lang.String, org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	
+	@Override
+	public void delete(String parentUniqueId, String uuid, String reason, RequestContext context) throws ResponseException {
+		if (parentUniqueId != null && uuid != null) {
+			// DELETE /patient/UUID/allergy/ALLERGYUUID means to delete one allergy from patient

```

Comment:
I am not sure what you mean.
As the code stands the call to the service layer will throw an exception which the web services framework will return.
Should I be checking to make sure the allergy exists before I try to delete?